src/text/line.rs

```diff
@@ -293,37 +295,47 @@ impl<'a> Line<'a> {
     ///
     /// assert_ne!(raw_line, styled_line);
     ///
-    /// raw_line.patch_style(style);
+    /// let raw_line = raw_line.patch_style(style);
     /// assert_eq!(raw_line, styled_line);
     /// ```
-    pub fn patch_style<S: Into<Style>>(&mut self, style: S) {
+    #[must_use = "method moves the value of self and returns the modified value"]
+    pub fn patch_style<S: Into<Style>>(mut self, style: S) -> Self {
         let style = style.into();
-        for span in &mut self.spans {
-            span.patch_style(style);
-        }
+        self.spans = self
+            .spans
+            .into_iter()
+            .map(|span| span.patch_style(style))
+            .collect();
+        self
```

Comment:
It's worth considering whether to change this to `self.style = self.style.patch(style)`
And likewise the same for reset, and Text patch / reset.

The impact of this is that the styles are not stored in the Span - this both a pro and con.

Reply:
I think `patch_style` on a struct should patch the style _of this struct_ and not its internal structs.

Won't be able to do that for Text though, as you noted in https://github.com/ratatui-org/ratatui/issues/758 it doesn't have style.

Reply:
yup. The grand plan is to gradually build that out so the hierarchy is right. :)